spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/engine/ClassParserUsingASM.java

```diff
@@ -490,6 +492,19 @@ public void visitEnd() {
             mBuilder.addParameterAnnotation(parameter, desc, value);
             return value.getAnnotationVisitor();
         }
+
+        @Override
+        public org.objectweb.asm.AnnotationVisitor visitTypeAnnotation(int typeRef, TypePath typePath,
+                String desc, boolean visible) {
+            TypeReference typeRefObject = new TypeReference(typeRef);
+            if (typeRefObject.getSort() == TypeReference.METHOD_FORMAL_PARAMETER) {
```

Comment:
It would be more readable if this were reversed: "if not the right sort then return `null`". No big deal.